### PR TITLE
Make project compile against the latest rust nightly

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2,9 +2,9 @@
 name = "foxbox"
 version = "0.1.0"
 dependencies = [
- "clippy 0.0.37 (registry+https://github.com/rust-lang/crates.io-index)",
+ "clippy 0.0.41 (registry+https://github.com/rust-lang/crates.io-index)",
  "docopt 0.6.78 (registry+https://github.com/rust-lang/crates.io-index)",
- "docopt_macros 0.6.80 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (JohanLorenzo) [inline Cargo.lock:7]: Changes in this file were automatically made by `cargo update`

File: src/events.rs [hunk 2]
```diff
@@ -16,8 +16,8 @@ impl EventData {
     pub fn description(&self) -> String {
         let description = match *self {
             EventData::AdapterStart { ref name } => name,
-            EventData::ServiceStart { ref id } => id,
-            EventData::ServiceStop { ref id } => id
+            EventData::ServiceStart { ref id }
+            | EventData::ServiceStop { ref id } => id,
```
Comment (arcturus) [inline src/events.rs:20]: Why is this change needed? Should be part of the update to nightly?

Comment (JohanLorenzo) [inline src/events.rs:20]: That's because the linter needed to be updated too. I dealt with this change in a separated commit https://github.com/JohanLorenzo/foxbox/commit/c78a39309f5a06a48af94992e73e36a3a377d7fa

**Comment by arcturus** [comment]
LGTM, travis green, merging.

**Comment by fabricedesre** [comment]
That's still not building for me, with rustc 1.8.0-dev (3a254fe4b 2016-02-15) :

/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:46:35: 46:44 error: use of undeclared type name `ast::Pat_` [E0412]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:46     pub fn build_pat_(self, pat_: ast::Pat_) -> F::Result {
                                                                                                                           ^~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:46:35: 46:44 help: run `rustc --explain E0412` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56:25: 56:43 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56         self.build_pat_(ast::Pat_::PatWild)
                                                                                                                 ^~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56:25: 56:43 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56:25: 56:43 error: unresolved name `ast::Pat_::PatWild` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56         self.build_pat_(ast::Pat_::PatWild)
                                                                                                                 ^~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:56:25: 56:43 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64:25: 64:44 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64         self.build_pat_(ast::Pat_::PatIdent(mode, id, sub))
                                                                                                                 ^~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64:25: 64:44 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64:25: 64:44 error: unresolved name `ast::Pat_::PatIdent` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64         self.build_pat_(ast::Pat_::PatIdent(mode, id, sub))
                                                                                                                 ^~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:64:25: 64:44 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170:33: 170:51 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170         self.builder.build_pat_(ast::Pat_::PatEnum(self.path, Some(self.pats)))
                                                                                                                          ^~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170:33: 170:51 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170:33: 170:51 error: unresolved name `ast::Pat_::PatEnum` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170         self.builder.build_pat_(ast::Pat_::PatEnum(self.path, Some(self.pats)))
                                                                                                                          ^~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:170:33: 170:51 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275:33: 275:53 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275         self.builder.build_pat_(ast::Pat_::PatStruct(self.path, self.pats, true))
                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275:33: 275:53 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275:33: 275:53 error: unresolved name `ast::Pat_::PatStruct` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275         self.builder.build_pat_(ast::Pat_::PatStruct(self.path, self.pats, true))
                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:275:33: 275:53 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279:33: 279:53 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279         self.builder.build_pat_(ast::Pat_::PatStruct(self.path, self.pats, false))
                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279:33: 279:53 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279:33: 279:53 error: unresolved name `ast::Pat_::PatStruct` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279         self.builder.build_pat_(ast::Pat_::PatStruct(self.path, self.pats, false))
                                                                                                                          ^~~~~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:279:33: 279:53 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314:27: 314:44 error: failed to resolve. Could not find `Pat_` in `syntax::ast` [E0433]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314         self.0.build_pat_(ast::Pat_::PatLit(expr))
                                                                                                                    ^~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314:27: 314:44 help: run `rustc --explain E0433` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314:27: 314:44 error: unresolved name `ast::Pat_::PatLit` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314         self.0.build_pat_(ast::Pat_::PatLit(expr))
                                                                                                                    ^~~~~~~~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:314:27: 314:44 help: run `rustc --explain E0425` to see a detailed explanation
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:345:33: 345:44 error: unresolved name `ast::PatTup` [E0425]
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:345         self.builder.build_pat_(ast::PatTup(self.pats))
                                                                                                                          ^~~~~~~~~~~
/home/fabrice/.cargo/registry/src/github.com-88ac128001ac3a9a/aster-0.12.0/src/pat.rs:345:33: 345:44 help: run `rustc --explain E0425` to see a detailed explanation

**Comment by arcturus** [comment]
Should we stay with the specific version from past friday?

**Comment by fabricedesre** [comment]
Yes.

**Comment by JohanLorenzo** [comment]
See #51 for pinning the rust compiler to a known working revision.
